lib/web_ui/lib/src/engine/pointer_binding/event_position_helper.dart

```diff
@@ -33,6 +43,24 @@ ui.Offset computeEventOffsetToTarget(DomMouseEvent event, DomElement actualTarge
   return ui.Offset(event.offsetX, event.offsetY);
 }
 
+/// Computes the offsets for input nodes, which live outside of the shadowDOM.
+/// Since inputs can be transformed (scaled, translated, etc), we can't rely on
+/// `event.offset` to be accurate. `_computeOffsetRelativeToActualTarget` only
+/// handles the case where inputs are translated, but will have issues for scaled
+/// inputs (see: https://github.com/flutter/flutter/issues/125948).
+///
+/// We compute the offsets here by using the text input geometry data that is
+/// sent from the framework, which includes information on how to transform the
+/// underlying input element. We transform the `event.offset` points we receive
+/// using the values from the input's transform matrix.
+ui.Offset _computeOffsetForInputs(DomMouseEvent event) {
```

Comment:
One more check we can do here is whether the event target is the same as the active input element i.e. `textEditing.strategy.domElement`.

Reply:
That's a good call.  Should we `assert` this or just put it in the outside conditional so the code has a chance to fall through to the fallback case?

Reply:
If the event target is different from the active input, we are going to produce the wrong offsets regardless of which method we use. So I don't care much about which method we use in this case (`_computeOffsetForInputs` or fallback). What I do care about is being able to notice when this unexpected scenario occurs, so IMHO, we should assert.

Reply:
This is updated btw